DirectEmployers/MyJobs
# Retrieve jobs from Redis/MySQL instead of Solr.

## What's this PR do?
Gets search results from Solr, and then pulls the actual jobs from redis, falling back to mysql if required.
Makes some changes to ensure that all required information to display for a job is actually available. 

## Where should the reviewer start?
Most of the new logic is in seo/utils/search.py

## How should this be manually tested?
Ask me for the new Solr schema (and how to replace it if you don't know that either). Testing this without the new Solr schema is useless, so don't bother unless you get that first.
Index a few jobs.
Browse around - confirm that nothing is missing (I'll be doing this with the DS team this week) and that search results make sense.
A few things to check - 
- job detail pages
- search results pages
- autocomplete on all three search boxes
- default and custom faceting


## Any background context you want to provide?
This should be the last major piece (there are still a few very minor pieces and some bug fixes coming) required for the search overhaul. 

There are still a few parts I'm not overly thrilled with, so I'm interested in, and would appreciate, feedback.
